pkg/controller/publishingstrategy/publishingstrategy_controller.go

```diff
@@ -105,78 +109,87 @@ func (r *ReconcilePublishingStrategy) Reconcile(request reconcile.Request) (reco
-	log.Info(fmt.Sprintf("appingress on publishingstrategy CR: %+v", instance.Spec.ApplicationIngress))
-	log.Info(fmt.Sprintf("ingress to reconcile: %+v", ingressNotOnCluster))
-	// -----------------------------------------------------------------------------------------------------------
-
-	for _, appingress := range ingressNotOnCluster {
-		newCertificate := &corev1.LocalObjectReference{
-			Name: appingress.Certificate.Name,
+		// Let an IngressController with a deletion timestamp delete
+		if !ingressController.DeletionTimestamp.IsZero() {
+			return reconcile.Result{Requeue: true}, nil
 		}
 
-		if appingress.Default == true {
-			err := r.defaultIngressHandle(appingress, ingressControllerList, newCertificate)
-			if err != nil {
-				log.Error(err, fmt.Sprintf("failed to handle default ingresscontroller %v", appingress))
-				return reconcile.Result{}, err
+		// Check Spec fields that cannot be patched vs desired IngressController
+		if !validateStaticSpec(*ingressController, desiredIngressController.Spec) {
+			// "default" CR also needs a status check
+			if ingressDefinition.Default {
+				if !validateStaticStatus(*ingressController, desiredIngressController.Spec) {
+					//Delete IngressController as it doesn't match ApplicationIngress
+					r.client.Delete(context.TODO(), ingressController)
+					return reconcile.Result{Requeue: true}, nil
+				}
+			} else {
+				//Delete IngressController as it doesn't match ApplicationIngress
+				r.client.Delete(context.TODO(), ingressController)
+				return reconcile.Result{Requeue: true}, nil
 			}
-			continue
-		}
-		err := r.nonDefaultIngressHandle(appingress, ingressControllerList, newCertificate)
-		if err != nil {
-			log.Error(err, fmt.Sprintf("failed to handle non-default ingresscontroller %v", appingress))
 		}
-	}
-
-	ingressList := &operatorv1.IngressControllerList{}
-	listOptions = []client.ListOption{
-		client.InNamespace("openshift-ingress-operator"),
-	}
-	err = r.client.List(context.TODO(), ingressList, listOptions...)
-	if err != nil {
-		log.Error(err, "Cannot get list of ingresscontroller")
-		return reconcile.Result{}, err
-	}
 
-	// since we don't own ingresscontroller object, adding this check ensures that the proper ingresscontrollers have been created by cluster-ingress-operator
-	// if check fails, requeue the reconcile and try to re-create the ingresscontrollers until successful
-	if !r.ensureIngressControllersExist(instance.Spec.ApplicationIngress, ingressList) {
-		reqLogger.Info("IngressController does not match PublishingStrategy. Requeue and try again")
-		return reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+		// Check Spec fields that CAN be patched vs desired IngressController
+		if valid, field := validatePatchableSpec(*ingressController, desiredIngressController.Spec); !valid {
+			baseToPatch := client.MergeFrom(ingressController.DeepCopy())
+			if ingressDefinition.Default {
```

Comment:
Out of curiosity why is there different behaviour between default ingress vs additional ones?

Reply:
"defult" CR also needs a status check because on initial creation the spec isn't filled out